src/Plugin/RetryPlugin.php

```diff
@@ -120,23 +142,35 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
             }
 
             $time = call_user_func($this->exceptionDelay, $request, $exception, $this->retryStorage[$chainIdentifier]);
-            usleep($time);
-
-            // Retry synchronously
-            ++$this->retryStorage[$chainIdentifier];
-            $promise = $this->handleRequest($request, $next, $first);
 
-            return $promise->wait();
```

Comment:
hm, did this ever work? this will return the actual response (unless there is an exception that is thrown, which also is wrong as it should be a rejected promise instead).

should we not call ourselves recursively instead?

Reply:
ah wait. we are inside the `then` of the promise, so this is the correct thing to do, i think.